File: SERVEUR/COMPILATEUR/mml-webapp/scripts_upload/mml.py
import pandas as pd
import numpy as np
from sklearn.model_selection import cross_validate
from sklearn.model_selection import RepeatedKFold
from sklearn.ensemble import RandomForestClassifier 

#Using pandas to import the dataset 
data = pd.read_csv('scripts_upload/upload/iris.csv', sep=',')
columns_names = data.columns

#Spliting dataset between features (X) and label (y)
Y_name = columns_names[-1]
Y = data[Y_name] 
X = data.drop(columns=Y_name) 


#Cross Validation
numRepetitionCross = 1
rkf = RepeatedKFold( n_repeats=numRepetitionCross)
results=[['Algorithm', 'Parameters', 'FrameWork', 'balanced_accuracy', 'macro_f1']]


try:
	algo = RandomForestClassifier()
	cv_results = cross_validate(algo, X, Y, cv= rkf, scoring=['balanced_accuracy', 'f1_macro'])
	framework_algo = ['RandomForest', '', 'scikit-learn']
	framework_algo.extend([cv_results[k].mean() for k in cv_results.keys() if k.startswith("test_")])
	results.append(framework_algo)
except: print('At least one algorithm has received an error')


pd.DataFrame(results).to_csv("scripts_upload/results/results_python.csv",  sep=";", header=None, index=None)

File: SERVEUR/COMPILATEUR/mml-webapp/scripts_upload/mml.R
packages<- function(name){
  if(eval(parse(text=paste("!require(",name,")")))){
    install.packages(name)
    eval(parse(text=paste("library(",name,")")))
  }
}
packages("yardstick")
packages("rminer")
packages("randomForest") 

#Import the dataset 
data = read.csv('scripts_upload/upload/iris.csv', sep=',')
columns_names = colnames(data)

#Spliting dataset between features (X) and label (y)
#y-> the last column
Y_name = rev(columns_names)[1]
X_names = rev(columns_names)[-1]
formula <- as.formula(paste(Y_name, "~", paste(X_names, collapse= "+"))) 



# Spliting dataset into training set and test set
train_size = 0.8
indice_train= sample(c(1:nrow(data)), train_size*nrow(data))
train=data[indice_train, ]
test=data[-indice_train, ]
Y_test = test[Y_name] 
X_test = test[X_names]
Y_train = train[Y_name] 
X_train = train[X_names]

results_colnames = c('Algorithm', 'Parameters', 'FrameWork', 'balanced_accuracy', 'accuracy')
results = matrix(ncol=length(results_colnames),nrow=1)

try({
	algo = randomForest(formula, data=train)
	Y_pred = predict(algo, X_test,  type="class")
	results = rbind(results, c('RandomForest', '', 'R', bal_accuracy_vec(Y_test[,1], Y_pred) 
	, accuracy_vec(Y_test[,1], Y_pred) 
	))
	
	
})

results_df = data.frame(results, row.names = NULL)
colnames(results_df) =results_colnames
results_df = results_df[-1,]
write.table(results_df,"scripts_upload/results/results_R.csv", sep = ";",dec='.',quote=FALSE, row.names = FALSE)
